protocol/communities_messenger_token_permissions_test.go

```diff
@@ -445,6 +445,97 @@ func (s *MessengerCommunitiesTokenPermissionsSuite) TestJoinedCommunityMembersSe
 	}
 }
 
+func (s *MessengerCommunitiesTokenPermissionsSuite) TestEditSharedAddresses() {
+	community, _ := s.createCommunity()
+	s.advertiseCommunityTo(community, s.alice)
+
+	s.joinCommunity(community, s.alice, alicePassword, []string{aliceAddress2})
+
+	community, err := s.owner.GetCommunityByID(community.ID())
+	s.Require().NoError(err)
+
+	s.Require().Equal(2, community.MembersCount())
+
+	for pubKey, member := range community.Members() {
+		if pubKey != common.PubkeyToHex(&s.owner.identity.PublicKey) {
+			s.Require().Len(member.RevealedAccounts, 1)
+
+			switch pubKey {
+			case common.PubkeyToHex(&s.alice.identity.PublicKey):
+				s.Require().Equal(member.RevealedAccounts[0].Address, aliceAddress2)
+			default:
+				s.Require().Fail("pubKey does not match expected keys")
+			}
+		}
+	}
+
+	passwdHash := types.EncodeHex(crypto.Keccak256([]byte(alicePassword)))
+	request := &requests.RequestToJoinCommunityOrEditSharedAddresses{CommunityID: community.ID(), Password: passwdHash, AddressesToReveal: []string{aliceAddress1}}
+	response, err := s.alice.EditSharedAddressesForCommunity(request)
+	s.Require().NoError(err)
+	s.Require().NotNil(response)
+
+	// Retrieve address change
+	err = tt.RetryWithBackOff(func() error {
+		response, err := s.owner.RetrieveAll()
+		if err != nil {
+			return err
+		}
+		if len(response.Communities()) == 0 {
+			return errors.New("no communities in response (address change reception)")
+		}
+		return nil
+	})
+	s.Require().NoError(err)
+
+	//Check that the owner has the new address
+	community, err = s.owner.GetCommunityByID(community.ID())
+	s.Require().NoError(err)
+
+	for pubKey, member := range community.Members() {
+		if pubKey != common.PubkeyToHex(&s.owner.identity.PublicKey) {
+			s.Require().Len(member.RevealedAccounts, 1)
+
+			switch pubKey {
+			case common.PubkeyToHex(&s.alice.identity.PublicKey):
+				s.Require().Equal(member.RevealedAccounts[0].Address, aliceAddress1)
```

Comment:
I found that the test works 90% of the time, but sometimes, it fails here. It seems to fail because the Owner didn't retrieve the message to edit the community. How can I fix that? cc @0x-r4bbit @mprakhov

Reply:
Hard to say without debugging :) 

I would suggest you check that you received the correct response first here https://github.com/status-im/status-go/pull/3723/files#diff-19758c8021697444e5216815e7a24d0537cd971a0018fc4b0116517b8930b961R479-R489

Also, the problem can be that before you get your edit event - you are receiving some other event and handling this event instead of your edit event

Reply:
Also, check `WaitOnMessengerResponse` fn, maybe it makes sense to reuse it as it uses merge

Reply:
Good tips, I'll try

Reply:
I managed to fix with your tip in the first comment. Thanks @mprakhov . 
I think you're right that another event came in first.